File: app/src/main/java/com/martin/preventapp/ui/new_order/Company/CompanyFragment.java
package com.martin.preventapp.ui.new_order.Company;


import android.os.Bundle;



import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.Spinner;
import android.widget.Toast;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import com.martin.preventapp.MainActivity;
import com.martin.preventapp.R;
import com.martin.preventapp.databinding.FragmentCompanyBinding;
import com.martin.preventapp.databinding.FragmentNewOrderBinding;
import com.martin.preventapp.firebase.Clients;
import com.martin.preventapp.firebase.Company;
import com.martin.preventapp.firebase.Products;
import com.martin.preventapp.ui.new_order.NewOrderFragment;

public class CompanyFragment extends Fragment {

    private FragmentCompanyBinding binding;
    private String CompanySelected;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        binding = FragmentCompanyBinding.inflate(inflater, container, false);
        View root = binding.getRoot();

        //Spinner company

        Company CompanyList = new Company();

        //Spinner Company

        Spinner spinnerCompany = root.findViewById(R.id.spinnerCompany);
        ArrayAdapter<String> adapter = new ArrayAdapter<String>(getActivity(), android.R.layout.simple_spinner_item, CompanyList.companyList(root));
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        spinnerCompany.setAdapter(adapter);

        spinnerCompany.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> adapterView, View view, int position, long l) {
                CompanySelected = adapterView.getItemAtPosition(position).toString();
                Toast.makeText(root.getContext(), CompanySelected, Toast.LENGTH_SHORT).show();
            }

            @Override
            public void onNothingSelected(AdapterView<?> adapterView) {

            }
        });

        //go to NewOrderFragment
        Button GoNewOrder = root.findViewById(R.id.go_order);

        Bundle bundle = new Bundle();

        FragmentManager fragmentManager = getParentFragmentManager();// getSupportFragmentManager();
        NewOrderFragment newOrderFragment = new NewOrderFragment();

        GoNewOrder.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                if(CompanySelected == "Seleccione un proveedor"){
                    Toast.makeText(root.getContext(), "Seleccione un proveedor", Toast.LENGTH_SHORT).show();
                }else {

                    bundle.putString("CompanySelected", CompanySelected); // Put anything what you want
                    newOrderFragment.setArguments(bundle);

                    fragmentManager.beginTransaction()
                    .replace(R.id.nav_host_fragment_content_main, newOrderFragment)
                    .addToBackStack(null) // name can be null
                    .commit();
                }
            }
        });

        return root;
    }
}

File: app/src/main/java/com/martin/preventapp/View/new_order/Company/CompanyFragment.kt
package com.martin.preventapp.View.new_order.Company

import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import android.widget.*
import com.martin.preventapp.firebase.Company
import com.martin.preventapp.R
import com.martin.preventapp.View.new_order.NewOrderFragment
import androidx.fragment.app.Fragment
import com.martin.preventapp.databinding.FragmentCompanyBinding

class CompanyFragment : Fragment() {
    private lateinit var binding: FragmentCompanyBinding
    private var CompanySelected: String = ""

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentCompanyBinding.inflate(layoutInflater)
        val root: View = binding.root

        //Spinner company
        val CompanyList = Company()

        //Spinner Company
        val adapter = ArrayAdapter(
            root.context,
            android.R.layout.simple_spinner_item,
            CompanyList.companyList(root)
        )
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        binding.spinnerCompany.adapter = adapter
        binding.spinnerCompany.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
            override fun onItemSelected(
                adapterView: AdapterView<*>,
                view: View,
                position: Int,
                l: Long
            ) {
                CompanySelected = adapterView.getItemAtPosition(position).toString()
            }

            override fun onNothingSelected(adapterView: AdapterView<*>) {}
        }

        //go to NewOrderFragment
        val bundle = Bundle()
        val fragmentManager = parentFragmentManager // getSupportFragmentManager();
        val newOrderFragment = NewOrderFragment()
        binding.goOrder.setOnClickListener {
            if (CompanySelected === "Seleccione un proveedor") {
                Toast.makeText(root.context, "Seleccione un proveedor", Toast.LENGTH_SHORT).show()
            } else {
                bundle.putString("CompanySelected", CompanySelected) // Put anything what you want
                newOrderFragment.arguments = bundle
                fragmentManager.beginTransaction()
                    .replace(R.id.nav_host_fragment_content_main, newOrderFragment)
                    .addToBackStack(null) // name can be null
                    .commit()
            }
        }
        return root
    }
}